drivers/volume/portworx/portworx.go

```diff
@@ -979,34 +990,22 @@ func (d *portworx) WaitDriverUpOnNode(n node.Node, timeout time.Duration) error
 
 func (d *portworx) WaitDriverDownOnNode(n node.Node) error {
 	t := func() (interface{}, bool, error) {
-		// to avoid getting the same node which driver has brought down
-		nManager, err := d.pickAlternateClusterManager(n)
-		if err != nil {
-			return "", true, &ErrFailedToWaitForPx{
-				Node:  n,
-				Cause: err.Error(),
-			}
-		}
-		nodeInspectResponse, err := nManager.Inspect(d.getContext(), &api.SdkNodeInspectRequest{NodeId: n.VolDriverNodeID})
-		if err != nil {
-			return "", true, &ErrFailedToWaitForPx{
-				Node:  n,
-				Cause: err.Error(),
-			}
-		}
-		if nodeInspectResponse.Node.Status != api.Status_STATUS_OFFLINE {
-			return "", true, &ErrFailedToWaitForPx{
-				Node: n,
-				Cause: fmt.Sprintf("px is not yet down on node. Expected: %v Actual: %v",
-					api.Status_STATUS_OFFLINE, nodeInspectResponse.Node.Status),
+
+		for _, addr := range n.Addresses {
+			err := d.testAndSetEndpointUsingNodeIP(addr)
+			if !strings.Contains(err.Error(), "connect: connection refused") {
```

Comment:
instead a string connection refused shouldn't we test for a proper status check? this check can hide networking issues for example